app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/MainViewModel.kt

```diff
@@ -84,6 +121,41 @@ class MainViewModel(
         }
     }
 
+    suspend fun fetchAllData(context: Context) {
+
+
+        val typesToFetch = allDataTypes
+//        val typesToFetch: List<DataType.ClinicalResource> = (0 until 100).flatMap {
+//            allDataTypes
+//        }
+
+        val startTime = Date()
+        Timber.d("Starting %d fetches", typesToFetch.count())
+
+        val jobs = typesToFetch.mapIndexed { index, clinicalResource ->
+            Timber.d("Launching coroutine %d", index)
+            CoroutineScope(Dispatchers.IO).launch {
+                Timber.d("co %d: Fetching %s results", index, clinicalResource.uniqueKey)
+                val results = fetchData(context, clinicalResource).mapNotNull { it as? ClinicalDataQueryResult }
+                Timber.d("co %d: Fetched %d %s results", index, results.count(), clinicalResource.uniqueKey)
+                resultsHolderActor!!.send(
+                    ResultHolderMessage.SetResults(
+                        clinicalResource,
+                        results
+                    )
+                )
+            }
+        }
+        jobs.joinAll()
+
+
+        val endTime = Date()
+        val totalDuration = (endTime.time - startTime.time).toDouble() / 1000.0
+        val avgDuration = totalDuration / typesToFetch.count()
```

Comment:
Do we want to leave the profiling code in this sample app, or the one in Android-PHR repo?

Reply:
Removed this